atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java

```diff
@@ -316,6 +320,58 @@ public CompletableFuture<ReconfigureResponse> onReconfigure(final ReconfigureReq
     }
   }
 
+  @Override
+  public CompletableFuture<ForceConfigureResponse> onForceConfigure(
+      final ForceConfigureRequest request) {
+    logRequest(request);
+    updateTermAndLeader(request.term(), null);
+
+    final var currentConfiguration = raft.getCluster().getConfiguration();
+
+    // No need to overwrite if already in force configuration. This can happen due to retry.
+    if (currentConfiguration == null
+        || !currentConfiguration.force()
+        || request.index() > currentConfiguration.index()) {
+      final var configurationIndex =
+          Math.max(request.index(), raft.getCurrentConfigurationIndex() + 1);
+      raft.getCluster()
+          .configure(
+              new Configuration(
+                  configurationIndex, // use the latest index instead of one from the request
+                  raft.getTerm(), // use the latest term instead of one from the request
+                  request.timestamp(),
+                  request.newMembers(),
+                  List.of(), // Skip joint consensus
+                  true));
+      raft.getCluster().commitCurrentConfiguration();
+    } else if (!(currentConfiguration.allMembers().containsAll(request.newMembers())
```

Comment:
:question: This part confused me a bit, mainly because of the additional `request.index() > currentConfiguration.index()` above. This implies there's some cases where we _are_ already in force phase, but can accept newer requests? So if we had a request with a greater index and different members, that would be fine? :thinking: 

So if I understand correctly, the expectation is if the request index is <= configuration index, this means the operation is not new but instead being retried, hence the issue of the members are different? Could they not be two concurrent ones but that stem from different nodes? Just a user error? I guess it's still valid of course :smile:

Reply:
:thinking: My thought was to cover a case where an out-dated follower did not come out of force config before the next force config. But as you said, this will prevent the detection of concurrent requests.

Reply:
I think it is better to remove this check. We have to figure out a different way to handle this out-dated follower case. May be use the term of the sender?

Reply:
I have removed this check and added a TODO to handle this case. I will address this in a follow up. In the current state, it will result in an endless retry. We would be able to recover from it by deleting the raft configuration file and restarting.